projects/dsp-ui/src/lib/viewer/views/resource-view/resource-view.component.ts

```diff
@@ -24,84 +33,95 @@ export interface PropertyInfoValues {
-  action: string; // label for the snackbar action
-
-  constructor(
-      @Inject(DspApiConnectionToken) private knoraApiConnection: KnoraApiConnection,
-      private _snackBar: MatSnackBar) { }
-
-  ngOnChanges() {
-    this.getResource(this.iri);
-  }
-
-  /**
-   * Get a read resource sequence with ontology information and incoming resources.
-   *
-   * @param iri resourceIri
-   */
-  getResource(iri: string): void {
-
-    this.knoraApiConnection.v2.res.getResource(iri).subscribe(
-      (response: ReadResource) => {
-        this.resource = response;
-
-        // gather resource property information
-        this.resPropInfoVals = this.resource.entityInfo.classes[this.resource.type].getResourcePropertiesList().map(
-            (prop: IHasPropertyWithPropertyDefinition) => {
-                const propInfoAndValues: PropertyInfoValues = {
-                    propDef: prop.propertyDefinition,
-                    guiDef: prop,
-                    values: this.resource.getValues(prop.propertyIndex)
-                };
-                return propInfoAndValues;
-            }
-        );
-
-        // sort properties by guiOrder
-        this.resPropInfoVals.sort((a, b) => (a.guiDef.guiOrder > b.guiDef.guiOrder) ? 1 : -1);
-
-        // get system property information
-        this.systemPropDefs = this.resource.entityInfo.getPropertyDefinitionsByType(SystemPropertyDefinition);
-
-        // set the arkUrl value
-        this.versionArkUrl = this.resource.versionArkUrl;
-
-      },
-      (error: ApiResponseError) => {
-        console.error('Error to get resource: ', error);
-      });
-  }
-
-  /**
-   * Display message to confirm the copy of the citation link (ARK URL)
-   * @param message
-   * @param action
-   */
-  openSnackBar(message: string, action: string) {
-    message = 'Copied to clipboard!';
-    action = 'Citation Link';
-    this._snackBar.open(message, action, {
-      duration: 3000,
-      horizontalPosition: 'center',
-      verticalPosition: 'top'
-    });
-  }
+export class ResourceViewComponent implements OnInit, OnChanges, OnDestroy {
+
+    /**
+     * Resource iri
+     *
+     * @param [iri] Resource iri
+     */
+    @Input() iri: string;
+
+    resource: ReadResource;
+
+    resPropInfoVals: PropertyInfoValues[] = []; // array of resource properties
+
+    systemPropDefs: SystemPropertyDefinition[] = []; // array of system properties
+
+    versionArkUrl: string; // versionArkUrl value
+    message: string; // message to show in the snackbar to confirm the copy of the ARK URL
+    action: string; // label for the snackbar action
+
+    eventBusSubscription: Subscription;
+
+    constructor(
+        @Inject(DspApiConnectionToken) private knoraApiConnection: KnoraApiConnection,
+        private _snackBar: MatSnackBar,
+        public _eventBusService: EventBusService) { }
+
+    ngOnInit() {
+        this.eventBusSubscription = this._eventBusService.on(Events.ValueAdded, () => this.getResource(this.iri));
```

Comment:
it would be great to have a little sentence to explain what is going on here

Reply:
explanation added in c212919